### PR TITLE
Add packer method for building centos 6.5 base amis

### PR DESCRIPTION
**Pull request by dividehex**
This commit adds a template and supporting resources in order to
        build a centos 6.5 base ami for use with puppet again.  This
        method is meant to replace previous method of using the aws_create_ami.py
        script.

        The outputed AMI will be named: centos65-hvm-64-base-{timestamp}

### CODE REVIEW

**Comment by dividehex** [comment]
I'll also note the entire drive behind this is to build an AMI that is not limited to a /boot partition.  This template creates a centos AMI with a single root partition that can be grown out to whatever size the ec2 instance is created at.

File: packer/src/install.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+CHROOT="/mnt/cloud_root"
+ROOT_DEV="/dev/xvdf"
+SRC="/tmp/src"
+
+yum -d 1 install -y MAKEDEV
+yum -d 1 install -y patch
+
+#parted -s $ROOT_DEV mktable gpt mkpart primary ext4 2048s 100%
+parted -s $ROOT_DEV mklabel gpt
```
Comment (rail) [inline packer/src/install.sh]: We used to use MBR in the past (aka msdos). This should work as well I guess.

File: packer/centos65-hvm-64-base.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            "device_name" : "/dev/xvdf",
+            "delete_on_termination" : false,
+            "volume_size" : 10
+            }
+        ],
+        "ami_root_device": {
+            "source_device_name": "/dev/xvdf",
+            "device_name": "/dev/xvda",
+            "delete_on_termination": true,
+            "volume_size": 16,
```
Comment (rail) [inline packer/centos65-hvm-64-base.json]: I completely forgot, do we need any resizing on boot or it's done by cloudinit?

Comment (dividehex) [reply packer/centos65-hvm-64-base.json]: I'm not entirely sure how the partition was previously expanded with the aws sciprts but in this case, when an instance is provisioned, the partition will need to be adjusted using growpart.  At the end of the provisioning, the instance is rebooted and cloud-init resizes the filesystem on startup.  I've been working on a terraform example that includes a short provisioning script to expand the partition when the instance is being created.

**Comment by dividehex** [comment]
> 
> 
> FTR, the idea behind separate `/boot` partition was to use as much instance storage as possible. The `/boot` partition had to be separate, so the `/` partition can use LVM and we just extend the disk size by extra EBS, that we attach.
> 
> This should simplify the initial boot process a lot (no need to do any LVM magic).

That makes sense.  I've assumed we are dropping support for instance storage and focusing entirely on EBS at this point.  This whole thing to me is just a stop gap until all the script worker apps in the aws account can be dockerized and migrated to cloud services.

**Pull request merged**